### ISSUE TITLE
Variation on White's first move

### ISSUE THREAD

**Issue by jmt1423** [shaack/cm-pgn#22]
Hey! Love the project and what I can do with it, I was just wondering if it is possible to add a variation on the first move of the game with pgn.history.addMove(). 

So as an example the variation tree should look like:

1. e4 (1.d4) e5

If it isn't possible I'd be happy to try and work on a solution and create a PR for it, but would like some clarification before I start trying a different solution. Thanks!

**Comment by shaack** [comment]
Hi @jmt1423. I'm not quite sure about that. I'll write a unit test right away to check this and give you feedback.

**Comment by shaack** [comment]
@jmt1423 You are right, it is not possible to a a variation on the first move with `addMove`, because, if you set the parameter `previous` to null, the method assumes, that you want to add the move at the end of the main history. I don't have a quick idea how to solve this, do you have one?

**Comment by shaack** [comment]
Shipped in v4.0.11. You can now add a variation on White's first move like this:

```js
const history = new History()
history.addMove("e4")
history.addMove("e5")
history.addMoveAtStart("d4")           // 1. e4 (1. d4) 1... e5
// or, equivalently:
history.addMove("d4", "start")
```

`addMoveAtStart` validates from the starting position (or from `setUpFen` if one is set), attaches the move as a variation on `moves[0]`, and returns the new move so you can extend the variation with further `addMove` calls. When the main line is empty, it simply becomes the first main-line move. Thanks for the patience!

**Comment by shaack** [comment]
Reverted the v4.0.11 release while a rendering instability triggered by the broader round-trip is sorted out. 4.0.11 is deprecated on npm; current released version is 4.0.10.

**Comment by shaack** [comment]
Shipped in v4.1.1.

```js
const history = new History()
history.addMove("e4")
history.addMove("e5")
history.addMoveAtStart("d4")     // 1. e4 (1. d4) 1... e5
// or, equivalently:
history.addMove("d4", "start")
```

`addMoveAtStart` validates from the starting position (or from `setUpFen`), attaches the move as a variation on `moves[0]`, and returns the new move so you can extend the variation with further `addMove` calls. When the main line is empty, it just becomes the first main-line move. Thanks for the patience!
